lib/logflare_web/controllers/plugs/verify_api_access.ex

```diff
@@ -18,9 +18,13 @@ defmodule LogflareWeb.Plugs.VerifyApiAccess do
   def call(conn, opts) do
     opts = Enum.into(opts, %{scopes: []})
     resource_type = Map.get(conn.assigns, :resource_type)
+    resource_owner = Map.get(opts, :resource_owner, Logflare.User)
```

Comment:
shouldnt need to do all these changes if we rely on scope.

can just do something like this in identify_requestor (simplified ver):
```
verify_func = case opts.scopes do
["partner"] -> :verify_partner_access_token
 _ -> :verify_access_token
end

with {:ok, user_or_partner} <- apply(Auth, verify_func, args) do
  {:ok, user_or_partner}
end 
```
then match on the struct to set assigns to correct key.